2352.ts

```diff
@@ -0,0 +1,14 @@
+function equalPairs(grid: number[][]): number {
+    let transposed: number[][] =  grid[0].map((col, c) => grid.map((row, r) => grid[r][c]))
```

Comment:
This is good stuff - some recommendations:
- `transposed` should be a `const` as it is only assigned once.
- `col` and `row` are unused parameters - the convention is to name them `_`
- A clarifying comment would be appropriate here.

Result:
```
// transpose the grid - from row x col, to col x 
const transposed: number[][] =  grid[0].map((_, col) => grid.map((_, row) => grid[col][row]))
```